lib/logstash/outputs/logstash.rb

```diff
@@ -122,15 +122,19 @@ def ssl_identity_options
       fail(LogStash::ConfigurationError, 'SSL identity can be configured with EITHER `ssl_certificate` OR `ssl_keystore_*`, but not both')
     elsif @ssl_certificate
       return {
-        'client_cert' => @ssl_certificate,
-        'client_key'  => @ssl_key || fail("`ssl_key` is REQUIRED when `ssl_certificate` is provided"),
+        'ssl_certificate' => @ssl_certificate,
+        'ssl_key'  => @ssl_key || fail(LogStash::ConfigurationError, "`ssl_key` is REQUIRED when `ssl_certificate` is provided"),
```

Comment:
I've mixed feelings about re-doing all these validations here. The HTTP input/output plugins are already validating them all, and should be the source of truth for all those validations. IMO, unless we have specific requirements for this plugin, we should just delegate validations to the under hood plugin. 
It's just a thought, I know your PR is not related/meant to change it :)

Reply:
Every layer (http input/output and mixin client) has its own validation processes and my opinion for this case would be to stick with validation (even extra) _to send meaningful parameters_. This would also hide internal implementation details that in the future we may move to other dependencies (ex: proposing to directly use http mixin instead http-output to support HA).

Reply:
I agree with you for the library's parameters/internal details, but considering this plugin is a kind of proxy to another user-facing plugin, I wouldn't bother validating the exposed settings again as the plugin should take care of it. I understand your point of sending meaningful parameters, but having, for example, different SSL validations on different plugins, has proved to lead to inconsistent behavior across them (to be addressed in the 2nd phase of the [SSL standardization project](https://github.com/elastic/logstash/issues/14905).

Anyway, thanks for your response, I just wanted to touch base on this topic as I've seen this same pattern in another plugin. We will probably be discussing it in the future :)